### PR TITLE
RHTAPSRE-467: Update application-service version

### PR DESCRIPTION
**Pull request by gbenhaim**
In order to get the changes introduced in
https://github.com/redhat-appstudio/application-service/pull/450, update the application service version.
In the new version some of the function moved between modules, so fixing the import paths for those.

This PR also locks the version of setup-envtest to overcome the issue described in https://github.com/kubernetes-sigs/controller-runtime/pull/2722

### CODE REVIEW

**Comment by gbenhaim** [comment]
envtest fails because of https://github.com/kubernetes-sigs/controller-runtime/pull/2722

File: Makefile [hunk 1]
```diff
@@ -167,7 +167,7 @@ kustomize: ## Download kustomize locally if necessary.
 ENVTEST = $(shell pwd)/bin/setup-envtest
 .PHONY: envtest
 envtest: ## Download envtest-setup locally if necessary.
-	$(call go-get-tool,$(ENVTEST),sigs.k8s.io/controller-runtime/tools/setup-envtest@latest)
+	$(call go-get-tool,$(ENVTEST),sigs.k8s.io/controller-runtime/tools/setup-envtest@v0.0.0-20240215143116-d0396a3d6f9f)
```
Comment (mmorhun) [inline Makefile:170]: What is the reason of this change?

File: go.mod [hunk 2]
```diff
@@ -27,18 +27,27 @@ require (
 	sigs.k8s.io/yaml v1.4.0
 )
 
+require (
+	github.com/devfile/alizer v1.2.2-0.20231004141146-f36141673c7f // indirect
```
Comment (mmorhun) [inline go.mod:31]: This should be in the already existing require section for indirect dependencies.

Comment (mmorhun) [reply go.mod:31]: I'll fix it later

**Review by mmorhun — APPROVED** [review]
